mimic/rest/auth_api.py

```diff
@@ -124,6 +124,17 @@ def get_username(self, request, tenant_id):
         session = self.core.sessions.session_for_tenant_id(tenant_id)
         return json.dumps(dict(user=dict(id=session.username)))
 
+    @app.route('/v2.0/users/<string:userid>/OS-KSADM/credentials/RAX-KSKEY:apiKeyCredentials',
+               methods=['GET'])
+    def rax_kskey_apikeycredentials(self, request, userid):
+        """
+        Support, such as it is, for the apiKeysCredentials call.
+        """
+        apikey = '1337133713371337133713371337'
+        username = 'mimicuser'
```

Comment:
You have to get the username from the session object. The session can be retrieved from `self.core.session_for_<something>`.  I think `userid` here is the same as `username`, so you could use the existing `session_for_username_password` method with a bogus password, or you could create a new `session_for_username` that explicitly doesn't authenticate anything.